pkg/types/dns/dns.go

```diff
@@ -0,0 +1,20 @@
+package dns
+
+// UserProvisionedDNS indicates whether the DNS solution is provisioned by the Installer or the user.
+type UserProvisionedDNS string
+
+const (
+	// UserProvisionedDNSEnabled indicates that the DNS solution is provisioned and provided by the user.
+	UserProvisionedDNSEnabled UserProvisionedDNS = "Enabled"
+
+	// UserProvisionedDNSDisabled indicates that the DNS solution is provisioned by the Installer.
+	UserProvisionedDNSDisabled UserProvisionedDNS = "Disabled"
+)
+
+// ValidUserProvisionedDNSType verifies the userProvisionedDNS string is valid
+func ValidUserProvisionedDNSType(userProvisionedDNS UserProvisionedDNS) bool {
+	if userProvisionedDNS == UserProvisionedDNSEnabled || userProvisionedDNS == UserProvisionedDNSDisabled {
+		return true
+	}
+	return false
+}
```

Comment:
This dns package change could be its own commit.

Reply:
Agree. I'll separate it.

Reply:
done.